src/dune_rules/mdx.ml

```diff
@@ -169,6 +173,11 @@ let decode =
        field "libraries" ~default:[]
          (Dune_lang.Syntax.since syntax (0, 2)
          >>> Dune_file.Lib_deps.decode Executable)
+     and+ locks =
+       field "locks"
+         (Dune_lang.Syntax.since syntax (0, 3)
+         >>> repeat String_with_vars.decode)
```

Comment:
It would be nice to share all the locks field related stuff between the stanzas. At least the field decoder for example

Reply:
OK I'll try something!